common/main/ai.h

```diff
@@ -99,7 +99,7 @@ namespace dsx {
 void create_awareness_event(vobjptr_t objp, player_awareness_type_t type);         // object *objp can create awareness of player, amount based on "type"
 ai_mode ai_behavior_to_mode(ai_behavior behavior);
 void do_ai_robot_hit(vobjptridx_t robot, player_awareness_type_t type);
-void init_ai_object(object &objp, ai_behavior initial_mode, segidx_t hide_segment);
+void init_ai_object(objptridx_t objp, ai_behavior initial_mode, segidx_t hide_segment);
```

Comment:
Instances of the base name wrap a value that is either None or a valid target.  The `c` prefix denies writing to the target, but still permits changing where the *X*__ptridx__ points.  The `v` prefix denies using None.  The `c` prefix is always compile-time checked, since it causes the contained pointer to be `const T *` instead of `T *`, so the compiler enforces the const-ness.  The `v` prefix is compile-time checked where possible (e.g. by having a `static_assert` in the implementation), but runtime-checked where necessary.  The `v` and `c` prefixes can be combined if the target is immutable and guaranteed not to be None.

```c++
    auto a = objptridx(i); // checks that i is -1 or a valid index, throws if neither
    vobjptridx_t b = a; // checks that a is a valid index, throws if not; if i is -1, this line throws
```